pkgs/by-name/rr/rrdtool/package.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, stdenv
+, fetchurl
+, gettext
+, perl
+, pkg-config
+, libxml2
+, pango
+, cairo
+, groff
+, tcl
+, darwin
+, nixosTests
+}:
+
+perl.pkgs.toPerlModule (stdenv.mkDerivation rec {
```

Comment:
Consider using the finalAttrs pattern, as rec is an anti-pattern.

Reply:
I'm not sure about that, I get the same nix store path with `rec` and with `finalAttrs`. And putting a trace for example `oldAttrs.outputs` in [toPerlModule](https://github.com/NixOS/nixpkgs/blob/49e703cde28bc7cbd3042d4709db9a21f9f84537/pkgs/top-level/perl-packages.nix#L37) yields the same result.

If you mean other people overriding this package, I'm not sure about that. (I'd still call myself a beginner in nixpkgs, sorry if I write anything stupid)